From patchwork Thu Nov 26 07:13:25 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: =?utf-8?b?5b2t5rWpKFJpY2hhcmQp?=
 <richard.peng@oppo.com>
X-Patchwork-Id: 11932763
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-13.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,INCLUDES_PATCH,
	MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS autolearn=unavailable
	autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 42C07C63777
	for <kvm@archiver.kernel.org>; Thu, 26 Nov 2020 07:14:15 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id DD3DF20DD4
	for <kvm@archiver.kernel.org>; Thu, 26 Nov 2020 07:14:14 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=oppoglobal.onmicrosoft.com
 header.i=@oppoglobal.onmicrosoft.com header.b="Se8K1To5"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S2388353AbgKZHNc (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 26 Nov 2020 02:13:32 -0500
Received: from mail-eopbgr1300070.outbound.protection.outlook.com
 ([40.107.130.70]:44583
        "EHLO APC01-HK2-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S2388333AbgKZHNa (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 26 Nov 2020 02:13:30 -0500
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=RtZ8yXx7cESq2gUAI/Ikjuof0EyfQM9mC2l5CjONk+Gq3jsWi0VddZh5nQer3RwaH5AicGLyi34zWpLx+zPQmq9p5Gh528EP5Y7vrImjaTXrQFRWqiz8xpwXNw6gvYoy42OlmS37wx6pf8QK7T/4d0E4gNV0yZO7c4JQzz5bQuAtK+VpwXSsDTk6kMCrhfuynwPY7eMcORMWREVGXv3Cwzu44cVr5eVC/AnyqibEdXBqdlcubmeFQu/sEUXYBmzC0inux0dSeFfXzycLi8nIQgbdXg5zpjXsoCrL7N874Zv5ft4hRD9G0FtpcoaS2lo/ddUcVHxtA6Oe+XxH+wTgaQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=+TKRrr3oyWGMVGmeIRJ2AOUfHaMeFfHrA8UgydINz24=;
 b=n41GqV4TpTJ/Dk28m3/RwgP1oBJIwDt3g2dZ+nUlgKoMFpcU/ejWv8GPgfyyBPiTCv5TBaqrQwRmGHavVSExEKzKGNT+RYD3R3c0aC3uMuuywrSiCvVrCzbc3VahR9FEwp3+J62zDDf6JkJFa9KO3yYjqvhKzuwrpWsB7MvmiMuHFNUWUA09jQ8WA3U9FqGmbg3ZuYDMEmKIgWxm+pn/0Pla6GIeXUu1dAoPvKXihIuyQTDNNTTdxZuvsCv0hajQFOnMFf990re7LB+VB1nmxUcACZljicSojaIOrf5Sj3C+2hL6s1ogE3ofRK05orVtnJi5NV2SMP/gta1FfcDgkQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oppo.com; dmarc=pass action=none header.from=oppo.com;
 dkim=pass header.d=oppo.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oppoglobal.onmicrosoft.com; s=selector1-oppoglobal-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=+TKRrr3oyWGMVGmeIRJ2AOUfHaMeFfHrA8UgydINz24=;
 b=Se8K1To5aHYtBDnSwsimwtWNO1CTllTyEO4Rgd2VaobzuX1GjiHJ76tf9jBJFXJmT1V73HuwFlYHwlMU0rrVwI5VOhnDf60EIxzuITAjfp2VYqypBbL14OQoPhnsyk8hRoqBeDkK+3pjrNCT+7n2YTSi8xc7W7krjok6AqN9Lcc=
Received: from HKAPR02MB4291.apcprd02.prod.outlook.com (2603:1096:203:d3::12)
 by HKAPR02MB4371.apcprd02.prod.outlook.com (2603:1096:203:d4::16) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3589.21; Thu, 26 Nov
 2020 07:13:25 +0000
Received: from HKAPR02MB4291.apcprd02.prod.outlook.com
 ([fe80::b9b8:aaf4:2afd:218b]) by HKAPR02MB4291.apcprd02.prod.outlook.com
 ([fe80::b9b8:aaf4:2afd:218b%3]) with mapi id 15.20.3611.022; Thu, 26 Nov 2020
 07:13:25 +0000
From: =?utf-8?b?5b2t5rWpKFJpY2hhcmQp?= <richard.peng@oppo.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
        "Suravee.Suthikulpanit@amd.com" <Suravee.Suthikulpanit@amd.com>
CC: "linux-kernel@vger.kernel.org" <linux-kernel@vger.kernel.org>,
        "kvm@vger.kernel.org" <kvm@vger.kernel.org>
Subject: [PATCH] kvm:svm: Return the correct error code
Thread-Topic: [PATCH] kvm:svm: Return the correct error code
Thread-Index: AdbDw07DFQcZfob5RjiYdtD4llKu/w==
Date: Thu, 26 Nov 2020 07:13:25 +0000
Message-ID: 
 <HKAPR02MB42915D77D43D4ED125BD2121E0F90@HKAPR02MB4291.apcprd02.prod.outlook.com>
Accept-Language: zh-CN, en-US
Content-Language: zh-CN
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
authentication-results: redhat.com; dkim=none (message not signed)
 header.d=none;redhat.com; dmarc=none action=none header.from=oppo.com;
x-originating-ip: [58.252.5.70]
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 6a599748-f468-40eb-8dc0-08d891dac444
x-ms-traffictypediagnostic: HKAPR02MB4371:
x-microsoft-antispam-prvs: 
 <HKAPR02MB437112ABEC0DFF8B82C3DD16E0F90@HKAPR02MB4371.apcprd02.prod.outlook.com>
x-ms-oob-tlc-oobclassifiers: OLM:2733;
x-ms-exchange-senderadcheck: 1
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 lhrudQuDxtDp4cnuUvn9SB2AK7ivEXxpELMwYIApoGSfXf556DifLdjoiPGpMdgQ5wRRqkuOQ+hLGkJeNSaR5cCYxanEnZb028LpfSWP0YA2wGf6+32PEseqYLwXK/AH0S06LyC6Fb1OOUVYiyvm9TAyFlREdqxTAyTdxO7ZvP5DTfdv/INJk50WYJZrXH7uzsjKc60VVW2OLvdz0x15TyHKw8do6zmAmHxJ0dmij9YG7wAbaJJ2BXuOdi5eAwDhZwJ7SURqWbqkPZDFF3MEHMFAnfhU6QMS/ve0lvZvixyn3SDpCKCciTTIsYPe2TDlOjtyi1sSx27G9f89jgcHjmt02fZixUzz5MQow1OW/OHPl7Xa6Ha89ArriUgN47Mi
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:HKAPR02MB4291.apcprd02.prod.outlook.com;PTR:;CAT:NONE;SFS:(4636009)(346002)(39860400002)(136003)(396003)(366004)(376002)(4744005)(6506007)(8676002)(83380400001)(4326008)(66946007)(52536014)(7696005)(8936002)(2906002)(86362001)(71200400001)(9686003)(55016002)(110136005)(85182001)(76116006)(66446008)(316002)(186003)(5660300002)(478600001)(66556008)(26005)(66476007)(64756008)(33656002)(54906003)(11606006);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata: =?utf-8?q?r2YzdCKsqn+V0OtINLRjqWS4rr20To?=
	=?utf-8?q?T/HWrWB+u5V/+wAgkScC0sWYLk0Vio7dMjFhxHMcygV8pSvVmMTHMfW8Hwzqvu4W9?=
	=?utf-8?q?UIJDb73D7U4V7Vm++jUKk/vQb1TpTHkmdy+1Vcx48jBMTom3ondPlCpHrHmcnsQks?=
	=?utf-8?q?hiJ16lbN+FbXhocRzys+UP06TDk7+G5BNUtIsxgv1r0+VKiUD/CoVcb0Zkn3QP65j?=
	=?utf-8?q?GX74sbM7z1SkeEfThhgoYZ0KBHW7fnuc8SKswf9XV0g0k6WkyW6kjph2uipZq8f6t?=
	=?utf-8?q?HFkLtgBa7QweGK8WVXrlEq3LpXiMaVsdiPn8hGuNzO8B6/ye63J8Ctylfhv6LbCSo?=
	=?utf-8?q?q6H5wpmDxsYiggMIZncKdAehroPY9Nhu2AkXUvT0j1aj4bmmx9mlF1tpkVcfpvHof?=
	=?utf-8?q?Bg6pC9abU1LsoIis6QtK2E7Oh2rety5fZoPabK0Jb3D2ztOEuhPM48wJZZcG6r18c?=
	=?utf-8?q?s4fxcGfqIJRNd8Icp5IApRMSjX9dTM9waJNLuaHCeeuSFvXCwNB7E4ofF3NHd6wMB?=
	=?utf-8?q?T81tJ5/W/1lu8OM419EdtSd5GJq7shGKaoxEvOP+2HE8pQHLUijfT2LbmrcsmM1uR?=
	=?utf-8?q?NeRDyU4KLK0GNOUSBB/FXT/3KzLzTuCg0ABx+tC10S+1yUIuvmIlZm6xyO/NP/+Fm?=
	=?utf-8?q?IOO8V7EaayrRUHr4QRWEcKnFWseqHdXl3AjssMIPuQbgPwiyMah2p0/S1QhxXjnqE?=
	=?utf-8?q?9sKh5xYNgFN0EyiJr7t4BE4JTvbcQfxNOS5FbdD31yn/JJWy9If+NcWhJpRsStcy/?=
	=?utf-8?q?pEF38u7+qUpcJdP1Use8AaxHI9uTP9BfuV0c14e6XL6mSdUgf94a6IhHwjTe3PzVF?=
	=?utf-8?q?63SpxLjbeBL0/MnQ1/VoaEoS1hKOHXr+A/2hu9193KwiCGC/M2dmflp3n0DbWZCNt?=
	=?utf-8?q?ZsM0LATZ7u58YUlhiO1TraREcmKVipPpx1wKq/F1J20iEcDW0SpmM8Jha6zCdfE?=
	=?utf-8?q?=3D?=
x-ms-exchange-transport-forked: True
MIME-Version: 1.0
X-OriginatorOrg: oppo.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: HKAPR02MB4291.apcprd02.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 6a599748-f468-40eb-8dc0-08d891dac444
X-MS-Exchange-CrossTenant-originalarrivaltime: 26 Nov 2020 07:13:25.3405
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: f1905eb1-c353-41c5-9516-62b4a54b5ee6
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 8022BFKxVRNZrz8zG//NzvZfzFjG65vKHQ+fua2finVIwUCeRgvpo0nliW8BlI28nTAj+36gP0jTpET1+QwOJw==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: HKAPR02MB4371
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The return value of sev_asid_new is assigned to the variable asid, which
should be returned directly if the asid is an error code.

Fixes: 1654efcbc431 ("KVM: SVM: Add KVM_SEV_INIT command")
Signed-off-by: Peng Hao <richard.peng@oppo.com>
---
 arch/x86/kvm/svm/sev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--
2.18.4

diff --git a/arch/x86/kvm/svm/sev.c b/arch/x86/kvm/svm/sev.c
index 566f4d18185b..41cea6b69860 100644
--- a/arch/x86/kvm/svm/sev.c
+++ b/arch/x86/kvm/svm/sev.c
@@ -174,7 +174,7 @@ static int sev_guest_init(struct kvm *kvm, struct kvm_sev_cmd *argp)

        asid = sev_asid_new();
        if (asid < 0)
-               return ret;
+               return asid;

        ret = sev_platform_init(&argp->error);
        if (ret)
